### ISSUE TITLE
Data compatibility for XiView

### ISSUE THREAD

**Issue by owenran** [Rappsilber-Laboratory/xiview#565]
I am using XiView to visualize a set of crosslinking experiments but am unable to visualize the 2D network(XiNet view).  I’ve attached an example of the data I am uploading; [WT_NDC80_DSSO_MQ_xiView_NoRepeat.csv](https://github.com/user-attachments/files/15980135/WT_NDC80_DSSO_MQ_xiView_NoRepeat.csv) and FASTA file(unable to attach). 

When uploaded to XiView, I do see the circular contact map: <img width="1440" alt="Xi_View_Circular" src="https://github.com/Rappsilber-Laboratory/xiview/assets/173854700/65aa6bfc-ec5b-4251-b507-92bfcae4cd7b"> & 
<img width="1440" alt="XiView_XiNetView_Null" src="https://github.com/Rappsilber-Laboratory/xiview/assets/173854700/305fd683-5314-41a4-9466-4fe926f8f838">
but am unable to visualize the 2D network.

Additionally, I’ve attempted uploading the data into XiNet directly(using the example documents to format my .csv file), but receive an error:
<img width="1440" alt="XiNet_error" src="https://github.com/Rappsilber-Laboratory/xiview/assets/173854700/98fc60b3-ea29-40aa-9a6b-61fdc0827d8c">

Any assistance or direction you can provide would be greatly appreciated.

**Comment by colin-combe** [comment]
Hi,
there is something wrong with the circle view you are seeing, also in the first screenshot i can see the sequence is missing for the protein WT, and in the the xiNET error message it is saying that it failed to find a sequence in a FASTA file.

So, i think the id's don't match up between the CSV and FASTA. 

If you rename the FASTA file so the file extension is .txt then you should be able to upload it here. If you do that, then I'll take a look at the problem.

cheers,
Colin

**Comment by owenran** [comment]
Hi Colin,
Thanks for checking this out, I've attached the FASTA file as a .txt.  

If it's naming convention or unallowed characters, I can go through the data and simplify the protein names/titles.

Best,
Randy
[WT_NDC80_DSSO.txt](https://github.com/user-attachments/files/15992550/WT_NDC80_DSSO.txt)

**Comment by colin-combe** [comment]
Hi Randy,

xiVIEW has a problem with the new lines between the '>' line and the line containing the sequence. I think it's technically allowed to have FASTA files like this, but the current version of xiVIEW doesn't like it.

I think if you edit the fasta file so it's like [WT_NDC80_DSSO.txt](https://github.com/user-attachments/files/16009847/WT_NDC80_DSSO.txt) then it will work.

Also, if you go to the 'DISPLAY' dropdown menu in top bar and uncheck 'crop at underscore' then you will be able to see the protein names properly.

Let me know if other you have any other problems,
cheers,
Colin

**Comment by owenran** [comment]
Hi Colin,

I made the changes you suggested, but now I am getting an error message (
<img width="1440" alt="Screenshot 2024-06-27 at 12 04 26 PM" src="https://github.com/Rappsilber-Laboratory/xiview/assets/173854700/ce2e5461-1025-403a-878b-efcdc122cbcf">), which, if I click the link at the bottom ("Try loading data only"), I see this screen (
<img width="1440" alt="Screenshot 2024-06-27 at 12 04 32 PM" src="https://github.com/Rappsilber-Laboratory/xiview/assets/173854700/108435f1-c82d-4d30-8997-a9056d44fa97">)

Additionally, if I exit out of the error message the only visualization I see is 
<img width="1440" alt="Screenshot 2024-06-27 at 12 08 12 PM" src="https://github.com/Rappsilber-Laboratory/xiview/assets/173854700/162fa3cb-0730-4c79-93d0-f019dec1715e"> without the linear/2D XiNet view.  

I also tried deleting all the spaces between sequences in the FASTA file, to the same result.  

However, the 'crop at underscore' option worked great!  Thank you.

Best,
Randy

**Comment by colin-combe** [comment]
hmmm, for me it worked. 
![Screenshot from 2024-06-27 20-08-50](https://github.com/Rappsilber-Laboratory/xiview/assets/977607/3653c445-59a7-4655-bee9-66c8bacf2fb2)

does this link work for you? : 
https://xiview.org/network.php?upload=28023-02488-29466-99123-63020
if not, what browser are you using?

if that link does work, you should be ok. You need to upload the data again, and then the page that leads to should work?

if not, is it the exact files above you tried to upload?

**Comment by owenran** [comment]
Hey Colin,

I don't know how you did it, but when I downloaded the same files as above, it worked.   Additionally, when I clicked the link, it worked.  

I am using Vivaldi (Chromium based) on a Mac.  

I thought it may have something to do with how I was processing the FASTA files, because I was generating them in the default TextEdit software on Apple computers, which outputs a text file as a .rtf, not plain .txt, which I then converted to FASTA files in Terminal.  I went back, recreated the FASTA files by first creating plain .txt files (instead of .rtf), then converted to .FASTA files and suddenly it worked.  

Thank you immensely!

Best,
Randy
